[BUG]: tests intermittently timing out robotInit

### Problem description

we saw this time out in GitHub Actions today

```
=================================== FAILURES ===================================
________________________________ test_disabled _________________________________
...
            # If your robotInit is taking more than 2 seconds in simulation, you're
            # probably doing something wrong... but if not, please report a bug!
>           assert self._cond.wait_for(lambda: self._robot_initialized, timeout=2)
E           AssertionError

.venv/lib/python3.14/site-packages/pyfrc/test_support/controller.py:75: AssertionError
```

https://github.com/thedropbears/pyrebuilt/actions/runs/21539711405/job/62072162572?pr=35#step:5:138

I don't think we're doing anything particularly unusual in our code

### Operating System

Linux

### Installed Python Packages

```text
Package                  Version
------------------------ ---------
attrs                    24.2.0
bcrypt                   4.2.0
cffi                     2.0.0
coverage                 7.13.1
cryptography             43.0.3
flexcache                0.3
flexparser               0.4
hypothesis               6.119.3
importlib-resources      6.5.2
iniconfig                2.0.0
librt                    0.7.8
mypy                     1.19.1
mypy-extensions          1.0.0
numpy                    2.3.5
opencv-python            4.10.0.84
packaging                23.2
paramiko                 3.5.0
pathspec                 1.0.3
phoenix6                 26.1.0
photonlibpy              2026.1.1
pint                     0.24.4
pip                      25.1.1
platformdirs             4.3.6
pluggy                   1.5.0
pycparser                2.22
pyfrc                    2026.0.1
pynacl                   1.5.0
pynetconsole             2.0.4
pyntcore                 2026.2.1
pytest                   8.3.3
pytest-integration       0.2.3
pytest-reraise           2.1.2
robotpy                  2026.2.1
robotpy-apriltag         2026.2.1
robotpy-cli              2026.0.1
robotpy-cscore           2026.2.1
robotpy-ctre             2026.1.0
robotpy-hal              2026.2.1
robotpy-halsim-gui       2026.2.1
robotpy-installer        2026.0.1
robotpy-native-apriltag  2026.2.1
robotpy-native-ntcore    2026.2.1
robotpy-native-wpihal    2026.2.1
robotpy-native-wpilib    2026.2.1
robotpy-native-wpimath   2026.2.1
robotpy-native-wpinet    2026.2.1
robotpy-native-wpiutil   2026.2.1
robotpy-rev              2026.0.1
robotpy-wpilib-utilities 2026.0.1
robotpy-wpimath          2026.2.1
robotpy-wpinet           2026.2.1
robotpy-wpiutil          2026.2.1
sleipnirgroup-choreolib  2026.0.1
sortedcontainers         2.4.0
tomli                    2.1.0
tomlkit                  0.13.2
typing-extensions        4.12.2
wpilib                   2026.2.1
```

### Reproducible example code

```text

```

--- a/pyfrc/mains/cli_test.py
+++ b/pyfrc/mains/cli_test.py
@@ -75,6 +75,12 @@ class PyFrcTest:
                 default=-1,
                 help="Maximum isolated robot processes (default: max CPUs - 1)",
             )
+            parser.add_argument(
+                "--init-timeout",
+                type=float,
+                default=None,
+                help="Seconds to wait for robot to start (can be set in `tool.robotpy.pyfrc.init_timeout` also)",
+            )
 
     def run(
         self,
@@ -87,6 +93,7 @@ class PyFrcTest:
         verbose: bool,
         pytest_args: typing.List[str],
         jobs: int,
+        init_timeout: typing.Optional[float],
     ):
         if isolated is None:
             pyproject_path = project_path / "pyproject.toml"
@@ -106,6 +113,26 @@ class PyFrcTest:
 
                     isolated = v
 
+                try:
+                    v = d["tool"]["robotpy"]["pyfrc"]["init_timeout"]
+                except KeyError:
+                    pass
+                else:
+                    if not isinstance(v, (int, float)):
+                        raise ValueError(
+                            f"tool.robotpy.pyfrc.init_timeout must be a number (got {v})"
+                        )
+                    elif not (v > 0):
+                        raise ValueError(
+                            f"tool.robotpy.pyfrc.init_timeout must be a positive number (got {v})"
+                        )
+
+                    if init_timeout is None:
+                        init_timeout = float(v)
+
+        if init_timeout is None:
+            init_timeout = 2.0
+
         if isolated is None:
             isolated = True
 
@@ -120,6 +147,7 @@ class PyFrcTest:
                 verbose,
                 pytest_args,
                 jobs,
+                init_timeout,
             )
         except _TryAgain:
             return self._run_test(
@@ -132,6 +160,7 @@ class PyFrcTest:
                 verbose,
                 pytest_args,
                 jobs,
+                init_timeout,
             )
 
     def _run_test(
@@ -145,6 +174,7 @@ class PyFrcTest:
         verbose: bool,
         pytest_args: typing.List[str],
         jobs: int,
+        init_timeout: float,
     ):
         # find test directory, change current directory so pytest can find the tests
         # -> assume that tests reside in tests or ../tests
@@ -180,14 +210,18 @@ class PyFrcTest:
                     pytest_args,
                     plugins=[
                         pytest_isolated_tests_plugin.IsolatedTestsPlugin(
-                            robot_class, main_file, builtin, verbose, jobs
+                            robot_class, main_file, builtin, verbose, jobs, init_timeout
                         )
                     ],
                 )
             else:
                 retv = pytest.main(
                     pytest_args,
-                    plugins=[pytest_plugin.PyFrcPlugin(robot_class, main_file, False)],
+                    plugins=[
+                        pytest_plugin.PyFrcPlugin(
+                            robot_class, main_file, False, init_timeout
+                        )
+                    ],
                 )
         finally:
             os.chdir(curdir)

--- a/pyfrc/test_support/controller.py
+++ b/pyfrc/test_support/controller.py
@@ -12,11 +12,12 @@ class TestController:
     Use this object to control the robot's state during tests
     """
 
-    def __init__(self, reraise, robot: wpilib.RobotBase):
+    def __init__(self, reraise, robot: wpilib.RobotBase, init_timeout: float = 2.0):
         self._reraise = reraise
 
         self._thread: typing.Optional[threading.Thread] = None
         self._robot = robot
+        self._init_timeout = init_timeout
 
         self._cond = threading.Condition()
         self._robot_started = False
@@ -72,7 +73,12 @@ class TestController:
 
             # If your robotInit is taking more than 2 seconds in simulation, you're
             # probably doing something wrong... but if not, please report a bug!
-            assert self._cond.wait_for(lambda: self._robot_initialized, timeout=2)
+            #
+            # To avoid this altogether, `--init-timeout` can be passed to `robotpy test`
+            # or you can add `tools.robotpy.pyfrc.init_timeout` instead
+            assert self._cond.wait_for(
+                lambda: self._robot_initialized, timeout=self._init_timeout
+            )
 
         try:
             # in this block you should tell the sim to do sim things

--- a/pyfrc/test_support/pytest_isolated_tests_plugin.py
+++ b/pyfrc/test_support/pytest_isolated_tests_plugin.py
@@ -128,7 +128,14 @@ class WorkerPlugin:
 
 
 def _run_test(
-    item_nodeid, config_args, robot_class, robot_file, verbose, pipe, root_path
+    item_nodeid,
+    config_args,
+    robot_class,
+    robot_file,
+    verbose,
+    pipe,
+    root_path,
+    init_timeout,
 ):
     """This function runs in a subprocess"""
     logging.root.addHandler(logging.NullHandler())
@@ -143,7 +150,7 @@ def _run_test(
 
     # keep the plugins around because it has a reference to the robot
     # and we don't want it to die and deadlock
-    plugin = PyFrcPlugin(robot_class, robot_file, True)
+    plugin = PyFrcPlugin(robot_class, robot_file, True, init_timeout)
     worker_plugin = WorkerPlugin(pipe)
 
     ec = pytest.main(
@@ -195,11 +202,13 @@ class IsolatedTestsPlugin:
         builtin_tests: bool,
         verbose: bool,
         parallelism: int,
+        init_timeout: float,
     ):
         self._robot_class = robot_class
         self._robot_file = robot_file
         self._builtin_tests = builtin_tests
         self._verbose = verbose
+        self._init_timeout = init_timeout
 
         if parallelism < 1:
             try:
@@ -287,6 +296,7 @@ class IsolatedTestsPlugin:
                 self._verbose,
                 cconn,
                 self._config.rootpath,
+                self._init_timeout,
             ),
         )
         process.start()

--- a/pyfrc/test_support/pytest_plugin.py
+++ b/pyfrc/test_support/pytest_plugin.py
@@ -36,6 +36,7 @@ class PyFrcPlugin:
         robot_class: Type[wpilib.RobotBase],
         robot_file: pathlib.Path,
         isolated: bool,
+        init_timeout: float,
     ):
         self.isolated = isolated
 
@@ -60,6 +61,8 @@ class PyFrcPlugin:
 
         self._robot_file = robot_file
         self._robot_class = TestRobot
+
+        self._init_timeout = init_timeout
 
         self._physics = physics
 
@@ -164,7 +167,7 @@ class PyFrcPlugin:
         """
         A pytest fixture that provides control over your robot
         """
-        return TestController(reraise, robot)
+        return TestController(reraise, robot, self._init_timeout)
 
     @pytest.fixture()
     def robot_file(self) -> pathlib.Path:

--- a/tests/test_pytest_plugins.py
+++ b/tests/test_pytest_plugins.py
@@ -75,7 +75,7 @@ from robot_module import {robot_class}
 
 def pytest_configure(config):
     robot_file = pathlib.Path(__file__).resolve()
-    config.pluginmanager.register(PyFrcPlugin({robot_class}, robot_file, False))
+    config.pluginmanager.register(PyFrcPlugin({robot_class}, robot_file, False, 2.0))
 """)
 
 
@@ -92,7 +92,7 @@ def pytest_configure(config):
         return
     robot_file = pathlib.Path(__file__).resolve()
     config.pluginmanager.register(
-        IsolatedTestsPlugin({robot_class}, robot_file, False, False, {parallelism})
+        IsolatedTestsPlugin({robot_class}, robot_file, False, False, {parallelism}, 2.0)
     )
 """)
 

